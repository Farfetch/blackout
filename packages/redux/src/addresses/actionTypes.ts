/**
 * Action type dispatched when the fetch prediction request fails.
 */
export const FETCH_ADDRESS_PREDICTIONS_FAILURE =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTIONS_FAILURE';
/**
 * Action type dispatched when the fetch prediction request starts.
 */
export const FETCH_ADDRESS_PREDICTIONS_REQUEST =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTIONS_REQUEST';
/**
 * Action type dispatched when the fetch prediction request succeeds.
 */
export const FETCH_ADDRESS_PREDICTIONS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTIONS_SUCCESS';

/**
 * Action type dispatched when the fetch prediction details request fails.
 */
export const FETCH_ADDRESS_PREDICTION_FAILURE =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTION_FAILURE';
/**
 * Action type dispatched when the fetch prediction details request starts.
 */
export const FETCH_ADDRESS_PREDICTION_REQUEST =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTION_REQUEST';
/**
 * Action type dispatched when the fetch prediction details request succeeds.
 */
export const FETCH_ADDRESS_PREDICTION_SUCCESS =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTION_SUCCESS';

/**
 * Action type dispatched when resetting the address prediction.
 */
export const RESET_ADDRESS_PREDICTIONS =
  '@farfetch/blackout-redux/RESET_ADDRESS_PREDICTIONS';
