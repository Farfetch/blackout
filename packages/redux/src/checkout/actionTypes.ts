/**
 * Action type dispatched when the create checkout order request fails.
 */
export const CREATE_CHECKOUT_ORDER_FAILURE =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_FAILURE';
/**
 * Action type dispatched when the create checkout order request starts.
 */
export const CREATE_CHECKOUT_ORDER_REQUEST =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_REQUEST';
/**
 * Action type dispatched when the create checkout order request succeeds.
 */
export const CREATE_CHECKOUT_ORDER_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_SUCCESS';

/**
 * Action type dispatched when the fetch checkout order charge request fails.
 */
export const FETCH_CHECKOUT_ORDER_CHARGE_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CHARGE_FAILURE';
/**
 * Action type dispatched when the fetch checkout order charge request starts.
 */
export const FETCH_CHECKOUT_ORDER_CHARGE_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CHARGE_REQUEST';
/**
 * Action type dispatched when the fetch checkout order charge request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_CHARGE_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CHARGE_SUCCESS';

/**
 * Action type dispatched when the create checkout charge request fails.
 */
export const CREATE_CHECKOUT_ORDER_CHARGE_FAILURE =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CHARGE_FAILURE';
/**
 * Action type dispatched when the create checkout order charge request starts.
 */
export const CREATE_CHECKOUT_ORDER_CHARGE_REQUEST =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CHARGE_REQUEST';
/**
 * Action type dispatched when the create checkout order charge request succeeds.
 */
export const CREATE_CHECKOUT_ORDER_CHARGE_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CHARGE_SUCCESS';
/**
 * Action type dispatched when resetting the checkout order charge state.
 */
export const RESET_CHECKOUT_ORDER_CHARGE_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_CHARGE_STATE';

/**
 * Action type dispatched when the fetch checkout request fails.
 */
export const FETCH_CHECKOUT_ORDER_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_FAILURE';
/**
 * Action type dispatched when the fetch checkout request starts.
 */
export const FETCH_CHECKOUT_ORDER_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_REQUEST';
/**
 * Action type dispatched when the fetch checkout request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order.
 */
export const RESET_CHECKOUT = '@farfetch/blackout-redux/RESET_CHECKOUT';

/**
 * Action type dispatched when the fetch checkout details request fails.
 */
export const FETCH_CHECKOUT_ORDER_DETAILS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DETAILS_FAILURE';
/**
 * Action type dispatched when the fetch checkout details request starts.
 */
export const FETCH_CHECKOUT_ORDER_DETAILS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DETAILS_REQUEST';
/**
 * Action type dispatched when the fetch checkout details request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DETAILS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DETAILS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order details state.
 */
export const RESET_CHECKOUT_ORDER_DETAILS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DETAILS_STATE';

/**
 * Action type dispatched when the fetch collectpoints request fails.
 */
export const FETCH_COLLECT_POINTS_FAILURE =
  '@farfetch/blackout-redux/FETCH_COLLECT_POINTS_FAILURE';
/**
 * Action type dispatched when the fetch collectpoints request starts.
 */
export const FETCH_COLLECT_POINTS_REQUEST =
  '@farfetch/blackout-redux/FETCH_COLLECT_POINTS_REQUEST';
/**
 * Action type dispatched when the fetch collectpoints request succeeds.
 */
export const FETCH_COLLECT_POINTS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_COLLECT_POINTS_SUCCESS';

/**
 * Action type dispatched when resetting the collect points state.
 */
export const RESET_COLLECT_POINTS_STATE =
  '@farfetch/blackout-redux/RESET_COLLECT_POINTS_STATE';

/**
 * Action type dispatched when the fetch item delivery provisioning request fails.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_FAILURE';
/**
 * Action type dispatched when the fetch item delivery provisioning request starts.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_REQUEST';
/**
 * Action type dispatched when the fetch item delivery provisioning request
 * succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order delivery bundle provisioning state.
 */
export const RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_STATE';

/**
 * Action type dispatched when the fetch delivery bundle upgrades request fails.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE';
/**
 * Action type dispatched when the fetch delivery bundle upgrades request starts.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST';
/**
 * Action type dispatched when the fetch delivery bundle upgrades request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order delivery bundle upgrades state.
 */
export const RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_STATE';

/**
 * Action type dispatched when the fetch item delivery provisioning request fails.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_FAILURE';
/**
 * Action type dispatched when the fetch item delivery provisioning request starts.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_REQUEST';
/**
 * Action type dispatched when the fetch item delivery provisioning request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order delivery bundle upgrade provisioning state.
 */
export const RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_STATE';

/**
 * Action type dispatched when the fetch order operation request fails.
 */
export const FETCH_CHECKOUT_ORDER_OPERATION_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATION_FAILURE';
/**
 * Action type dispatched when the fetch order operation request starts.
 */
export const FETCH_CHECKOUT_ORDER_OPERATION_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATION_REQUEST';
/**
 * Action type dispatched when the fetch order operation request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_OPERATION_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATION_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order operation state.
 */
export const RESET_CHECKOUT_ORDER_OPERATION_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_OPERATION_STATE';

/**
 * Action type dispatched when the fetch order operations request fails.
 */
export const FETCH_CHECKOUT_ORDER_OPERATIONS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATIONS_FAILURE';
/**
 * Action type dispatched when the fetch order operations request starts.
 */
export const FETCH_CHECKOUT_ORDER_OPERATIONS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATIONS_REQUEST';
/**
 * Action type dispatched when the fetch order operations request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_OPERATIONS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATIONS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order operations state.
 */
export const RESET_CHECKOUT_ORDER_OPERATIONS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_OPERATIONS_STATE';

/**
 * Action type dispatched when the remove checkout order item request fails.
 */
export const REMOVE_CHECKOUT_ORDER_ITEM_FAILURE =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_ITEM_FAILURE';
/**
 * Action type dispatched when the remove checkout order item request starts.
 */
export const REMOVE_CHECKOUT_ORDER_ITEM_REQUEST =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_ITEM_REQUEST';
/**
 * Action type dispatched when the remove checkout order item succeeds.
 */
export const REMOVE_CHECKOUT_ORDER_ITEM_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_ITEM_SUCCESS';

/**
 * Action type dispatched when resetting the remove checkout order item state.
 */
export const RESET_REMOVE_CHECKOUT_ORDER_ITEM_STATE =
  '@farfetch/blackout-redux/RESET_REMOVE_CHECKOUT_ORDER_ITEM_STATE';

/**
 * Action type dispatched when the set item tags request fails.
 */
export const SET_CHECKOUT_ORDER_ITEM_TAGS_FAILURE =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_ITEM_TAGS_FAILURE';
/**
 * Action type dispatched when the set item tags request starts.
 */
export const SET_CHECKOUT_ORDER_ITEM_TAGS_REQUEST =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_ITEM_TAGS_REQUEST';
/**
 * Action type dispatched when the set item tags request succeeds.
 */
export const SET_CHECKOUT_ORDER_ITEM_TAGS_SUCCESS =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_ITEM_TAGS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order item tags state.
 */
export const RESET_CHECKOUT_ORDER_ITEM_TAGS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_ITEM_TAGS_STATE';

/**
 * Action type dispatched when the set checkout order promocodes request fails.
 */
export const SET_CHECKOUT_ORDER_PROMOCODES_FAILURE =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_PROMOCODES_FAILURE';
/**
 * Action type dispatched when the set checkout order promocodes request starts.
 */
export const SET_CHECKOUT_ORDER_PROMOCODES_REQUEST =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_PROMOCODES_REQUEST';
/**
 * Action type dispatched when the set checkout order promocodes request succeeds.
 */
export const SET_CHECKOUT_ORDER_PROMOCODES_SUCCESS =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_PROMOCODES_SUCCESS';

/**
 * Action type dispatched when the remove checkout order promocodes request fails.
 */
export const REMOVE_CHECKOUT_ORDER_PROMOCODES_FAILURE =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_PROMOCODES_FAILURE';
/**
 * Action type dispatched when the remove checkout order promocodes request starts.
 */
export const REMOVE_CHECKOUT_ORDER_PROMOCODES_REQUEST =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_PROMOCODES_REQUEST';
/**
 * Action type dispatched when the remove checkout order promocodes request succeeds.
 */
export const REMOVE_CHECKOUT_ORDER_PROMOCODES_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_PROMOCODES_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order promocodes state.
 */
export const RESET_CHECKOUT_ORDER_PROMOCODES_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_PROMOCODES_STATE';

/**
 * Action type dispatched when the set tags request fails.
 */
export const SET_CHECKOUT_ORDER_TAGS_FAILURE =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_TAGS_FAILURE';
/**
 * Action type dispatched when the set tags request starts.
 */
export const SET_CHECKOUT_ORDER_TAGS_REQUEST =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_TAGS_REQUEST';
/**
 * Action type dispatched when the set tags request succeeds.
 */
export const SET_CHECKOUT_ORDER_TAGS_SUCCESS =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_TAGS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order tags state.
 */
export const RESET_CHECKOUT_ORDER_TAGS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_TAGS_STATE';

/**
 * Action type dispatched when the update checkout request fails.
 */
export const UPDATE_CHECKOUT_ORDER_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_FAILURE';
/**
 * Action type dispatched when the update checkout request starts.
 */
export const UPDATE_CHECKOUT_ORDER_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_REQUEST';
/**
 * Action type dispatched when the update checkout request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_SUCCESS';

/**
 * Action type dispatched when the update checkout order item request fails.
 */
export const UPDATE_CHECKOUT_ORDER_ITEM_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEM_FAILURE';
/**
 * Action type dispatched when the update checkout order item request starts.
 */
export const UPDATE_CHECKOUT_ORDER_ITEM_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEM_REQUEST';
/**
 * Action type dispatched when the update checkout order item request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_ITEM_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEM_SUCCESS';

/**
 * Action type dispatched when resetting the update checkout order item state.
 */
export const RESET_UPDATE_CHECKOUT_ORDER_ITEM_STATE =
  '@farfetch/blackout-redux/RESET_UPDATE_CHECKOUT_ORDER_ITEM_STATE';

/**
 * Action type dispatched when the update delivery bundle upgrade request fails.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_FAILURE';
/**
 * Action type dispatched when the update delivery bundle upgrade request starts.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_REQUEST';
/**
 * Action type dispatched when the update delivery bundle upgrade request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_SUCCESS';

/**
 * Action type dispatched when the update delivery bundle upgrades request fails.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE';
/**
 * Action type dispatched when the update delivery bundle upgrades request starts.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST';
/**
 * Action type dispatched when the update delivery bundle upgrades request
 * succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS';

/**
 * Action type dispatched when the update gift message request fails.
 */
export const UPDATE_CHECKOUT_ORDER_ITEMS_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEMS_FAILURE';
/**
 * Action type dispatched when the update gift message request starts.
 */
export const UPDATE_CHECKOUT_ORDER_ITEMS_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEMS_REQUEST';
/**
 * Action type dispatched when the update gift message request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_ITEMS_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEMS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order items state.
 */
export const RESET_CHECKOUT_ORDER_ITEMS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_ITEMS_STATE';

/**
 * Action type dispatched when the fetch payment methods request fails.
 */
export const FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_FAILURE';
/**
 * Action type dispatched when the fetch payment methods request starts.
 */
export const FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_REQUEST';
/**
 * Action type dispatched when the fetch payment methods request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_SUCCESS';

/**
 * Action type dispatched when the create checkout order context request fails.
 */
export const CREATE_CHECKOUT_ORDER_CONTEXT_FAILURE =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CONTEXT_FAILURE';
/**
 * Action type dispatched when the create checkout order context request starts.
 */
export const CREATE_CHECKOUT_ORDER_CONTEXT_REQUEST =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CONTEXT_REQUEST';
/**
 * Action type dispatched when the create checkout order context request succeeds.
 */
export const CREATE_CHECKOUT_ORDER_CONTEXT_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CONTEXT_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order contexts state.
 */
export const RESET_CHECKOUT_ORDER_CONTEXTS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_CONTEXTS_STATE';

/**
 * Action type dispatched when the fetch checkout order contexts request fails.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXTS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_FAILURE';
/**
 * Action type dispatched when the fetch checkout order contexts request starts.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXTS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_REQUEST';
/**
 * Action type dispatched when the fetch checkout order contexts request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order context state.
 */
export const RESET_CHECKOUT_ORDER_CONTEXT_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_CONTEXT_STATE';
/**
 * Action type dispatched when the fetch checkout order context request fails.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXT_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXT_FAILURE';
/**
 * Action type dispatched when the fetch checkout order context request starts.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXT_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXT_REQUEST';
/**
 * Action type dispatched when the fetch checkout order context request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXT_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXT_SUCCESS';

/**
 * Action type dispatched when the remove checkout order context request fails.
 */
export const REMOVE_CHECKOUT_ORDER_CONTEXT_FAILURE =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_CONTEXT_FAILURE';
/**
 * Action type dispatched when the remove checkout order context request starts.
 */
export const REMOVE_CHECKOUT_ORDER_CONTEXT_REQUEST =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_CONTEXT_REQUEST';
/**
 * Action type dispatched when the remove checkout order context request succeeds.
 */
export const REMOVE_CHECKOUT_ORDER_CONTEXT_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_CONTEXT_SUCCESS';
