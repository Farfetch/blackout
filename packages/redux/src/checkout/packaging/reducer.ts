import * as actionTypes from './../actionTypes.js';
import { type AnyAction, combineReducers } from 'redux';
import { createReducerWithResult } from '../../helpers/reducerFactory.js';
import type { PackagingOptionsState } from './index.js';

export const INITIAL_STATE: PackagingOptionsState = {
  error: null,
  result: null,
  isLoading: false,
};

export const packagingOptions = createReducerWithResult(
  'FETCH_PACKAGING_OPTIONS',
  INITIAL_STATE,
  actionTypes,
);

const error = (
  state = INITIAL_STATE.error,
  action: AnyAction,
): PackagingOptionsState['error'] => {
  switch (action.type) {
    case actionTypes.FETCH_PACKAGING_OPTIONS_FAILURE:
      return action.payload.error;
    case actionTypes.FETCH_PACKAGING_OPTIONS_REQUEST:
      return INITIAL_STATE.error;
    default:
      return state;
  }
};

const isLoading = (
  state = INITIAL_STATE.isLoading,
  action: AnyAction,
): PackagingOptionsState['isLoading'] => {
  switch (action.type) {
    case actionTypes.FETCH_PACKAGING_OPTIONS_REQUEST:
      return true;
    case actionTypes.FETCH_PACKAGING_OPTIONS_SUCCESS:
    case actionTypes.FETCH_PACKAGING_OPTIONS_FAILURE:
      return INITIAL_STATE.isLoading;
    default:
      return state;
  }
};

const result = (
  state = INITIAL_STATE.result,
  action: AnyAction,
): PackagingOptionsState['result'] => {
  switch (action.type) {
    case actionTypes.FETCH_PACKAGING_OPTIONS_SUCCESS:
      return action.payload.result;
    default:
      return state;
  }
};

export const getPackagingOptions = (
  state: PackagingOptionsState,
): PackagingOptionsState => state;

const packagingOptionsReducer = combineReducers({
  result,
  error,
  isLoading,
});

export default packagingOptionsReducer;
