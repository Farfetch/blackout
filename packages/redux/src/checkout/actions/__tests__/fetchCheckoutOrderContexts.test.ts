import * as actionTypes from '../../actionTypes.js';
import {
  checkoutOrderId,
  mockGetCheckoutOrderContextsResponse,
} from 'tests/__fixtures__/checkout/index.mjs';
import { fetchCheckoutOrderContexts } from '../index.js';
import { find } from 'lodash-es';
import { getCheckoutOrderContexts } from '@farfetch/blackout-client';
import { INITIAL_STATE } from '../../reducer.js';
import { mockStore } from '../../../../tests/index.js';

jest.mock('@farfetch/blackout-client', () => ({
  ...jest.requireActual('@farfetch/blackout-client'),
  getCheckoutOrderContexts: jest.fn(),
}));

describe('fetchCheckoutOrderContext() action creator', () => {
  const checkoutMockStore = (state = {}) =>
    mockStore({ checkout: INITIAL_STATE }, state);
  const expectedConfig = undefined;
  let store: ReturnType<typeof checkoutMockStore>;

  beforeEach(() => {
    jest.clearAllMocks();
    store = checkoutMockStore();
  });

  it('should create the correct actions for when the fetch checkout order contexts procedure fails', async () => {
    const expectedError = new Error('fetch checkout order context error');

    (getCheckoutOrderContexts as jest.Mock).mockRejectedValueOnce(
      expectedError,
    );

    await expect(
      async () =>
        await fetchCheckoutOrderContexts(checkoutOrderId)(store.dispatch),
    ).rejects.toThrow(expectedError);

    expect(getCheckoutOrderContexts).toHaveBeenCalledTimes(1);
    expect(getCheckoutOrderContexts).toHaveBeenCalledWith(
      checkoutOrderId,
      expectedConfig,
    );
    expect(store.getActions()).toEqual(
      expect.arrayContaining([
        { type: actionTypes.FETCH_CHECKOUT_ORDER_CONTEXTS_REQUEST },
        {
          type: actionTypes.FETCH_CHECKOUT_ORDER_CONTEXTS_FAILURE,
          payload: { error: expectedError },
        },
      ]),
    );
  });

  it('should create the correct actions for when the fetch checkout order contexts procedure is successful', async () => {
    (getCheckoutOrderContexts as jest.Mock).mockResolvedValueOnce(
      mockGetCheckoutOrderContextsResponse,
    );

    await fetchCheckoutOrderContexts(checkoutOrderId)(store.dispatch).then(
      clientResult => {
        expect(clientResult).toStrictEqual(
          mockGetCheckoutOrderContextsResponse,
        );
      },
    );

    const actionResults = store.getActions();

    expect(getCheckoutOrderContexts).toHaveBeenCalledTimes(1);
    expect(getCheckoutOrderContexts).toHaveBeenCalledWith(
      checkoutOrderId,
      expectedConfig,
    );

    expect(actionResults).toMatchObject([
      { type: actionTypes.FETCH_CHECKOUT_ORDER_CONTEXTS_REQUEST },
      {
        type: actionTypes.FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS,
        payload: {
          entities: {},
          result: mockGetCheckoutOrderContextsResponse.map(
            context => context.id,
          ),
        },
      },
    ]);
    expect(
      find(actionResults, {
        type: actionTypes.FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS,
      }),
    ).toMatchSnapshot('fetch checkout order context success payload');
  });
});
