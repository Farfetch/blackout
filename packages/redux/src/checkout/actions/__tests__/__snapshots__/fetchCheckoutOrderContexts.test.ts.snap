// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchCheckoutOrderContext() action creator should create the correct actions for when the fetch checkout order contexts procedure is successful: fetch checkout order context success payload 1`] = `
Object {
  "payload": Object {
    "entities": Object {
      "checkoutOrderContexts": Object {
        "3fa85f64-5717-4562-b3fc-2c963f66aaaa": Object {
          "context": "context2",
          "createdAt": "2023-05-16T17:54:29.565Z",
          "id": "3fa85f64-5717-4562-b3fc-2c963f66aaaa",
          "value": "value2",
        },
        "3fa85f64-5717-4562-b3fc-2c963f66afa6": Object {
          "context": "context1",
          "createdAt": "2023-05-05T17:54:29.565Z",
          "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
          "value": "value1",
        },
      },
    },
    "result": Array [
      "3fa85f64-5717-4562-b3fc-2c963f66afa6",
      "3fa85f64-5717-4562-b3fc-2c963f66aaaa",
    ],
  },
  "type": "@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS",
}
`;
